app/Http/Controllers/AdminController.php

```diff
@@ -1484,4 +1485,116 @@ private function gitinfo_file_contents(string $filename): string
 
         return $html;
     }
+
+    /**
+     * TODO: (williamjallen) Rewrite this to take advantage of Blade templating.
+     */
+    public function generateCTestConfig(): View|RedirectResponse|\Illuminate\Http\Response
+    {
+        @set_time_limit(0);
+
+        @$projectid = $_GET['projectid'];
+
+        // Checks
+        if (!isset($projectid) || !is_numeric($projectid)) {
+            return view('cdash', [
+                'xsl' => true,
+                'xsl_content' => 'Not a valid projectid!',
+                'title' => 'CTest Configuration'
+            ]);
+        }
+
+        $db = Database::getInstance();
+        $project_array = $db->executePreparedSingleRow('SELECT * FROM project WHERE id=?', [intval($projectid)]);
+        if (empty($project_array)) {
+            return view('cdash', [
+                'xsl' => true,
+                'xsl_content' => 'Not a valid projectid!',
+                'title' => 'CTest Configuration'
+            ]);
+        }
+        $policy = checkUserPolicy($project_array['id']);
+
+        if ($policy !== true) {
+            return $policy;
+        }
+
+        $ctestconfig = "## This file should be placed in the root directory of your project.\n";
+        $ctestconfig .= "## Then modify the CMakeLists.txt file in the root directory of your\n";
+        $ctestconfig .= "## project to incorporate the testing dashboard.\n";
+        $ctestconfig .= "##\n";
+        $ctestconfig .= "## # The following are required to submit to the CDash dashboard:\n";
+        $ctestconfig .= "##   ENABLE_TESTING()\n";
+        $ctestconfig .= "##   INCLUDE(CTest)\n";
+        $ctestconfig .= "\n";
+
+        $ctestconfig .= 'set(CTEST_PROJECT_NAME "' . $project_array['name'] . "\")\n";
+        $ctestconfig .= 'set(CTEST_NIGHTLY_START_TIME "' . $project_array['nightlytime'] . "\")\n\n";
+
+        if (config('app.env') === 'production') {
+            $ctestconfig .= "set(CTEST_DROP_METHOD \"https\")\n";
+        } else {
+            $ctestconfig .= "set(CTEST_DROP_METHOD \"http\")\n";
+        }
+
+        $ctestconfig .= 'set(CTEST_DROP_SITE "' . $_SERVER['SERVER_NAME'] . "\")\n";
```

Comment:
@williamjallen, would it be more useful if this were a query to the config to get the value of `$APP_URL` instead of using `SERVER_NAME`? 

Using `docker-compose` in a local instance puts out an "incorrect" file for my host system.  This value is written out as  `localhost` when the CDash instance is accessed at `localhost:8080`

Reply:
I was trying to avoid making changes beyond just moving the files around to keep things readable if we need to track down an issue in the future.  This change seems relatively straightforward through, so I'll push the changes momentarily.  The drop method is also incorrect, since many systems will use https.  I'll fix that as well.